lib/capistrano/s3/publisher.rb

```diff
@@ -189,17 +192,26 @@ def self.has_gzipped_version?(file)
           File.exist?(self.gzip_name(file))
         end
 
-        def self.build_gzip_content_type_hash(file, mime_type)
+        def self.build_gzip_content_type_hash(file, mime_type, prefer_cf_mime_types)
           orig_name = self.orig_name(file)
-          orig_mime = mime_type_for_file(orig_name)
+          orig_mime = mime_type_for_file(orig_name, prefer_cf_mime_types)
 
           return {} unless orig_mime && File.exist?(orig_name)
 
           { :content_type => orig_mime.content_type }
         end
 
-        def self.mime_type_for_file(file)
+        def self.mime_type_for_file(file, prefer_cf_mime_types)
           type = MIME::Types.type_for(file)
+
+          if prefer_cf_mime_types
+            intersection = type & Capistrano::S3::MIMETypes::CF_MIME_TYPES
+
+            if !intersection.empty?
+              type = intersection
```

Comment:
I would assign first type here 
```suggestion
              type = intersection.first
```